src/components/location/location.edgedb.repository.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@nestjs/common';
+import {
+  ID,
+  NotImplementedException,
+  PaginatedListType,
+  PublicOf,
+  UnsecuredDto,
+} from '~/common';
+import { e, RepoFor } from '~/core/edgedb';
+import { CreateLocation, Location, LocationListInput } from './dto';
+import { LocationRepository } from './location.repository';
+
+@Injectable()
+export class LocationEdgeDBRepository
+  extends RepoFor(Location, {
+    hydrate: (loc) => ({
+      ...loc['*'],
+      fundingAccount: true,
+      defaultFieldRegion: true,
+      mapImage: true,
+      defaultMarketingRegion: false, //TODO - add this to hydrate
+    }),
+  }).customize((cls) => {
+    return class extends cls {
+      async create(input: CreateLocation) {
+        const created = e.insert(e.Location, {
+          name: input.name,
+          type: input.type,
+          isoAlpha3: input.isoAlpha3,
+          fundingAccount:
+            input.fundingAccountId &&
+            e.cast(e.FundingAccount, e.cast(e.uuid, input.fundingAccountId)),
+          defaultFieldRegion:
+            input.defaultFieldRegionId &&
+            e.cast(e.FieldRegion, e.cast(e.uuid, input.defaultFieldRegionId)),
+          mapImage:
+            input.mapImage && e.cast(e.File, e.cast(e.uuid, input.mapImage)),
```

Comment:
@CarsonF - I'm unsure about how to handle this with the File type?

Reply:
Yeah TBD. You could leave a TODO until I figure it out.